test/Tools/LeanCode.CodeAnalysis.Tests/Analyzers/EnumDtoConsistencyAnalyzerTests.cs

```diff
@@ -0,0 +1,370 @@
+
+public enum Status
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 3
+}
+
+public enum StatusDTO
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2
+}";
+
+        var diags = new[] { new DiagnosticResult(DiagnosticsIds.EnumDtoShouldMatchBaseEnum, 10, 12) };
+        await VerifyDiagnostics(source, diags);
+    }
+
+    [Fact]
+    public async Task Enum_dto_with_exclude_members_attribute_should_pass()
+    {
+        var source =
+            AttributeDefinitions
+            + @"
+
+namespace Test;
+
+public enum Status
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 3
+}
+
+[LeanCode.CodeAnalysis.ExcludeMembers(Status.Completed, Status.Cancelled)]
+public enum StatusDTO
+{
+    None = 0,
+    InProgress = 1,
+}";
+
+        await VerifyDiagnostics(source);
+    }
+
+    [Fact]
+    public async Task Enum_dto_with_extra_members_should_fail()
+    {
+        var source =
+            @"namespace Test;
+
+public enum Status
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 3
+}
+
+public enum StatusDTO
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 3,
+    Deleted = 4
+}";
+
+        var diags = new[] { new DiagnosticResult(DiagnosticsIds.EnumDtoShouldMatchBaseEnum, 10, 12) };
+        await VerifyDiagnostics(source, diags);
+    }
+
+    [Fact]
+    public async Task Enum_dto_with_ignore_attribute_on_extra_member_should_pass()
+    {
+        var source =
+            AttributeDefinitions
+            + @"
+
+namespace Test;
+
+public enum Status
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 3
+}
+
+public enum StatusDTO
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 3,
+    [LeanCode.CodeAnalysis.IgnoreEnumValue]
+    Deleted = 4
+}";
+
+        await VerifyDiagnostics(source);
+    }
+
+    [Fact]
+    public async Task Enum_dto_with_value_mismatch_should_fail()
+    {
+        var source =
+            @"namespace Test;
+
+public enum Status
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 3
+}
+
+public enum StatusDTO
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 4
+}";
+
+        var diags = new[] { new DiagnosticResult(DiagnosticsIds.EnumDtoShouldMatchBaseEnum, 10, 12) };
+        await VerifyDiagnostics(source, diags);
+    }
+
+    [Fact]
+    public async Task Enum_dto_with_corresponds_attribute_should_pass()
+    {
+        var source =
+            AttributeDefinitions
+            + @"
+
+namespace Test;
+
+public enum Status
+{
+    None = 0,
+    InProgress = 1,
+    Completed = 2,
+    Cancelled = 3
+}
+
+public enum StatusDTO
+{
+    None = 0,
+    [LeanCode.CodeAnalysis.EnumValueCorresponds(Status.InProgress)]
+    InProgress = 2,
+    [LeanCode.CodeAnalysis.EnumValueCorresponds(Status.Completed)]
+    Completed = 3,
+    [LeanCode.CodeAnalysis.EnumValueCorresponds(Status.Cancelled)]
```

Comment:
Do remember that this is something you'll never see in practice. Contracts projects won't have references do domain projects, so at best you can do something like

```suggestion
    [LeanCode.CodeAnalysis.EnumValueCorresponds(3)]
```